File: packages/python/src/trend_buddy_shared/contracts/domain/marketdata.py
from dataclasses import dataclass
from datetime import datetime
from decimal import Decimal

from .instrument import Timeframe


@dataclass
class CandleKey:
    instrument_id: str
    timeframe: Timeframe


@dataclass
class Candle:
    instrument_id: str
    timeframe: Timeframe
    timestamp: datetime
    open: Decimal
    high: Decimal
    low: Decimal
    close: Decimal
    volume: Decimal


@dataclass
class Quote:
    instrument_id: str
    timestamp: datetime
    bid_price: Decimal
    ask_price: Decimal
    bid_size: Decimal = Decimal("0")
    ask_size: Decimal = Decimal("0")

File: packages/go/contracts/domain/marketdata.go
package domain

import "time"

type CandleKey struct {
	InstrumentID InstrumentID `json:"instrument_id"`
	Timeframe    Timeframe    `json:"timeframe"`
}

type Candle struct {
	InstrumentID InstrumentID `json:"instrument_id"`
	Timeframe    Timeframe    `json:"timeframe"`
	Timestamp    time.Time    `json:"timestamp"`
	Open         Decimal      `json:"open"`
	High         Decimal      `json:"high"`
	Low          Decimal      `json:"low"`
	Close        Decimal      `json:"close"`
	Volume       Decimal      `json:"volume"`
}

type Quote struct {
	InstrumentID InstrumentID `json:"instrument_id"`
	Timestamp    time.Time    `json:"timestamp"`
	BidPrice     Decimal      `json:"bid_price"`
	AskPrice     Decimal      `json:"ask_price"`
	BidSize      Decimal      `json:"bid_size,omitempty"`
	AskSize      Decimal      `json:"ask_size,omitempty"`
}
